assets/semgrep_rules/client/webui-origin-validation.yaml

```diff
@@ -0,0 +1,43 @@
+rules:
+  - id: chromium-webui-origin-validation
+    metadata:
+      author: Andrea Brancaleoni <abc@pompel.me>
+      references:
+        - https://github.com/brave/brave-browser/wiki/Security-reviews
+        - https://chromium.googlesource.com/chromium/src/+/main/docs/security/web-platform-security-guidelines.md
+      source: https://github.com/brave/security-action/blob/main/assets/semgrep_rules/client/webui-origin-validation.yaml
+      assignees: |
+        thypon
+        diracdeltas
+        fmarier
+      category: security
+    languages: [cpp, c]
+    message: |
+      WebUI pages (chrome:// and chrome-untrusted://) that handle external data
+      should validate the origin of requests. WebUI components have elevated
+      privileges and can be targets for cross-origin attacks. Always validate
+      that requests come from expected origins.
+    severity: WARNING
+    patterns:
+      - patterns:
+          - pattern-either:
+              - pattern-regex: void\s+Handle\w*\([^)]*std::string[^)]*\)
```

Comment:
this does not seem like an appropriate regex, `Handle` can be used in all kinds of different contexts and this is generating a lot of false positives already. Also please get @brave/brave-core-owners approval before adding chromium specific rulesets. Also I think ideally we should move these rules into brave-core if possible